src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs

```diff
@@ -290,6 +323,8 @@ public SymbolTable GetMutableCopyOfFunctions()
 
         private protected readonly TexlFunctionSet _functions = new TexlFunctionSet();
 
+        private protected readonly Dictionary<DName, FormulaType> _namedTypes = new Dictionary<DName, FormulaType>();

```

Comment:
>_namedTypes [](http://example.com/codeflow?start=66&length=11)

Can we actually move this to the derived class? 
(SymbolTable.cs)? 
(_variables probably ought to be there too, so that's  a bad pattern to follow...)

ReadOnlySymbolTable is an abstract base class. 
And we can make readers here abstract.

Reply:
Have made some changes , does this look okay ?